src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -721,9 +721,7 @@ impl<'a> TypeChecker<'a> {
             } => {
                 let func_name = normalize_identifier(name, self.name_resolution_ctx).to_string();
                 let func_name = func_name.as_str();
```

Comment:
what about handling case here?

```suggestion
                let func_name = func_name.to_lowercase();
```

Reply:
Good suggestion. Since sugar and built-in functions are matched ignoring case, how about we move the login inside the functions, otherwise each call site we have to convert to lower case before calling the functions

Reply:
Ok. I see. Then I'll suggest replacing `&str` in `all_sugar_functions` and `builtin_functions` to `https://crates.io/crates/unicase`

Reply:
I've used the Ascii type from the unicase crate, a case-insensitive wrapper for ASCII strings. Is this assumption correct? If not, I can switch to the UniCase type, which provides a case-insensitive wrapper for general strings.